src/providers/vmware/amd64.rs

```diff
@@ -39,8 +57,160 @@ impl VmwareProvider {
     fn fetch_guestinfo(erpc: &mut vmw_backdoor::EnhancedChan, key: &str) -> Result<Option<String>> {
         let guestinfo = erpc
             .get_guestinfo(key.as_bytes())
-            .context("failed to retrieve network kargs")?
+            .context(format!("failed to retrieve guestinfo for {}", key))?
             .map(|bytes| String::from_utf8_lossy(&bytes).into_owned());
         Ok(guestinfo)
     }
+
+    pub fn parse_netplan_config(&self) -> Result<Option<netplan_types::NetplanConfig>> {
+        if let Some(metadata) = &self.guestinfo_metadata {
+            // To provide at least some sanity checking, parse the netplan config.
+            // First check if we have JSON userdata.
+            let netplan_config: netplan_types::NetplanConfig =
+                if let Ok(netplan_json_config) = serde_json::from_str(metadata) {
+                    netplan_json_config
+                } else {
+                    // The JSON parsing error is discarded,
+                    // and if YAML parsing fails as well, we only return this error.
+                    serde_yaml::from_str(metadata)?
+                };
+            // Also check the version to make it easier for the OS to say that
+            // this mechanism is supported or not.
+            if netplan_config.network.version != 2 {
+                bail!("only netplan version 2 supported");
+            }
```

Comment:
I don't know much about Netplan though that seems out of scope for Afterburn. Why not be version agnostic and let the OS deal with version compatibility?

Reply:
The OS would run afterburn with this flag and then netplan with a particular version. If any netplan data can come in I thought it's better to narrow the supported values and to error out early - also, the netplan_types crate may not support newer versions.

Reply:
From previous comments:
> The distro would be responsible for configuring Netplan to translate to networkd or NetworkManager as desired.

I don't think we should parse the netplan config in Afterburn. We should leave that to another unit in the initrd or the root to do the translation for the desired backend.

Reply:
If the OS wants to error out before getting all the way to Netplan generation, integrators can ship a systemd service that runs right `After` Afterburn to check the config version.

Reply:
And actually, it seems like the only requirement on `netplan-types`? So if we drop this function, we should be able to get rid of the feature gating.

Reply:
Makes sense to me! :+1:

Reply:
And we also had the idea that `netplan-types` could be used in the providers for a NetworkManager-capable alternative to the `--network-units=` flag that Flatcar uses. If that needs arises we can still add it and it's not really needed at the moment.

Reply:
`local-hostname` is another top-level entry and not part of `netplan-types` because it's a cloud-init thing. Supporting and parsing that would be done with a separate struct.